wled00/FX_2Dfcn.cpp

```diff
@@ -513,6 +541,33 @@ void Segment::drawLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint3
   }
 }
 
+//line function
+void Segment::drawLineAntialiased(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint32_t c) {
+  if (!isActive()) return; // not active
+  const uint16_t cols = virtualWidth();
+  const uint16_t rows = virtualHeight();
+  if (x0 >= cols || x1 >= cols || y0 >= rows || y1 >= rows) return;
+  const int16_t dx = abs(x1-x0), sx = x0<x1 ? 1 : -1;
```

Comment:
Try to use `roundf()` or `lround()` -  it could improve accuracy and make the expression simpler because you don't need special handling for x1<x0

Reply:
Not sure I follow.

Reply:
sorry my mistake. I was assuming that `x0` and `x1` are float.